crates/chronicle/src/bootstrap/mod.rs

```diff
@@ -489,6 +493,33 @@ where
             .await?;
 
         Ok((response, ret_api))
+    } else if let Some(matches) = matches.subcommand_matches("perftest") {
+        let mut ops = matches.value_of("ops").unwrap().parse::<u64>().unwrap();
+        info!("Performing {} operations", ops);
+        let perftest_ops_api = api.clone();
+        let start_time = Instant::now();
+
+        tokio::task::spawn(async move {
+            while ops > 0 {
+                let identity = AuthId::chronicle();
+                let namespace = system_namespace();
+                let _res = perftest_ops_api.handle_perftest(identity, namespace).await;
+                ops -= 1;
+            }
+        });
+
+        let mut tx_notifications = api.notify_commit.subscribe();
```

Comment:
Should we subscribe _before_ we spawn the other task?

Reply:
It will subscribe from the stored block height:

```
       let (from_block_id, from_position) = {
            match &from_block_id {
                FromBlock::First => {
                    let first_block = self.get_first_block().await?;
                    debug!(first_block = %first_block);
                    (first_block, None)
                }
                FromBlock::Head => {
                    let (block_num, block_id) = self.get_block_height().await?;
                    debug!(block_height = %block_num, block_id = %block_id);
                    (block_id, Some(block_num))
                }
                FromBlock::BlockId(block_id) => (*block_id, None),
            }
        };
        ```

Reply:
And this may explain worse behaviour with restarts under load

Reply:
Ah, the `last_seen_block` in `Api::new`? :+1: